### PR TITLE
Bind specific null sql type from java.sql.Types.

### LINKED ISSUE #15871: Custom Views: setNull with correct SQL type on Bind
The `Bind.Option` and `Bind.Optional` currently use `setNull(pos, java.sql.Types.NULL)` in `None` and `empty()` case.
If the `Bind` would provide  the `java.sql.Types.X` code for the type it represents, we could use `setNull(pos, typeCode)` instead.
Not all JDBC drivers support `setNull(pos, java.sql.Types.NULL)`.

### PR DESCRIPTION
**Pull request by ray-roestenburg-da**
Fixes https://github.com/digital-asset/daml/issues/15871

### CODE REVIEW

File: src/test/scala/com/daml/projection/javadsl/JdbcPerfSpec.scala [hunk 1]
```diff
@@ -175,7 +169,7 @@ class JdbcPerfSpec
   def mkChoice = {
     Projection[ExercisedEvent](
       projectionId,
-      ProjectionFilter.parties(Set(partyIdBob))
+      ProjectionFilter.templateIdsByParty(Map(partyIdBob -> Set(templateId)))
```
Comment (ray-roestenburg-da) [inline src/test/scala/com/daml/projection/javadsl/JdbcPerfSpec.scala:172]: The perf test did not work without this change FYI @chunlokling-da

File: src/main/scala/com/daml/projection/JdbcAction.scala [hunk 2]
```diff
@@ -184,6 +184,24 @@ trait Bind[-T] {
...
+   * The [[java.sql.Types]] code that should be used when invoking `setNull` on `PreparedStatement`. Default in this
+   * trait is set to java.sql.Types.NULL, override if a more specific code is needed.
+   */
+  def nullSqlType: Int = java.sql.Types.NULL
+
+  /**
+   * Return a new Bind with the [[java.sql.Types]] code modified, that should be used when invoking `setNull` on
+   * `PreparedStatement`.
+   */
+  def withSqlType(sqlType: Int): Bind[T] = new Bind[T] {
```
Comment (cbley-da) [inline src/main/scala/com/daml/projection/JdbcAction.scala]: Perhaps a better name would be `withNullSqlType`.

**Review by cbley-da — APPROVED** [review]
LGTM, I would only rename this method.

File: src/main/scala/com/daml/projection/JdbcAction.scala (hunk 2 above)
Comment (ray-roestenburg-da) [reply src/main/scala/com/daml/projection/JdbcAction.scala]: Thanks.

Comment (ray-roestenburg-da) [reply src/main/scala/com/daml/projection/JdbcAction.scala]: Done.

**Review by S11001001 — APPROVED** [review]

**Pull request merged**